careers/careers/views.py

```diff
@@ -12,22 +15,37 @@ def home(request):
 
 def listings(request):
     return render(request, 'careers/listings.html', {
-        'positions': Position.objects.order_by('category__name', 'title'),
+        'positions': utils.get_all_positions(
+            order_by=lambda x: '{} {}'.format(x.category.name, x.title)),
         'form': PositionFilterForm(request.GET or None),
     })
 
 
 def position(request, job_id=None):
-    try:
-        position = Position.objects.select_related('category').get(job_id__contains=job_id)
-        positions = position.category.position_set.all()
-
-        # Add applicant source param for jobvite
-        position.apply_url += '&s=PDN'
-
-        return render(request, 'careers/position.html', {
-            'position': position,
-            'positions': positions,
-        })
-    except Position.DoesNotExist:
-        raise Http404
+    # Cannot use __exact instead of __contains due to MySQL collation
```

Comment:
Ha yes, the joys of mysql. I updated to comment with more information. I was trying to understand why edna had this as `__contains` instead of `__exact` and the reason is that the way mysql is configured for careers does not allow case sensitive matching. Using `__contains` works around that limitation.